server/server.js

```diff
@@ -41,6 +43,23 @@ if (process.env.ENV === "PRODUCTION") {
   });
 }
 
+app.get("/authenticate/user-posts", (req, res) => {
+  const { ghUsername, githubId, avatarUrl } = req.query;
+  User.find({ username: ghUsername })
+    .then(async (result) => {
+      if (result.length == 0) {
+        const newUser = new User({ githubId, username: ghUsername, avatarUrl });
+        const savedUser = await newUser.save();
+      }
+      return Post.find({ $or: [{ creator: ghUsername }, { isPublic: true }] });
+    })
+    .then((posts) => res.send(posts))
```

Comment:
As mentioned in the comment above, we're missing the user's `pinnedPosts` because we never returned the User data from the DB. If we don't go with the new approach (of instantiating/fetching the user in `/authenticate`) then we will need to return it here.

Reply:
Might also be good to sort by descending timestamp